### DISCUSSION TITLE
Datetime model field as separate date and time form fields

### DISCUSSION THREAD

**Discussion by axelbdt** [filamentphp/filament#15130] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.129

### How can we help you?

Hello,

I have a model with a datetime field.

I would like the form to have a date field and a time field, that are combined into the datetime field of my model.

I am not sure how to keep this idiomatic. Has anyone done this before?

Thanks!

**Comment by leandrocfe** [comment, marked as answer]
[Filament hooks](https://filamentphp.com/docs/3.x/panels/resources/creating-records/#customizing-data-before-saving)


[Laravel Accessor/Mutator](https://laravel.com/docs/11.x/eloquent-mutators#accessors-and-mutators)

**Reply by axelbdt** [reply]
Thank you for the pointers :)
